src/CoreShop2VueStorefrontBundle/DependencyInjection/CompilerPass/RepositoryProviderCompilerPass.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CoreShop2VueStorefrontBundle\DependencyInjection\CompilerPass;
+
+use CoreShop2VueStorefrontBundle\Bridge\RepositoryProvider;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Exception\LogicException;
+use Symfony\Component\DependencyInjection\Reference;
+
+class RepositoryProviderCompilerPass implements CompilerPassInterface
+{
+    private const REPOSITORY = 'coreshop2vuestorefront.repository';
+
+    private static $defaults = [
+        'coreshop.repository.category' => 'category',
```

Comment:
This could be moved to config so we allow the user to easily register additional repositories or change the default ones.